pilot/pkg/xds/endpoint_builder.go

```diff
@@ -285,6 +285,10 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 			if !subsetLabels.SubsetOf(ep.Labels) {
 				continue
 			}
+			// If we don't know the address we must eventually use a gateway address
```

Comment:
In this case the entry represents something in the proxy's own network. In that case, we don't use a gateway so this feature doesn't really make sense. We wouldn't have any endpoint for which this would contribute weight unless we know it's address.

Reply:
ah I missed the same network part